tests/extensions/image/test_exporter.py

```diff
@@ -0,0 +1,239 @@
+        assert output_image.palette == source_image.palette
+        assert output_image.format.lower() == settings.EXPORT_TYPE
+        for i in range(100):
+            # Check if conversion is close OR if the value on the source is 0. A value on the source
+            # Being zero means it was most likely transparent, and got converted to white.
+            assert isclose(source_pixels[i][0], output_pixels[i][0], abs_tol=5) or (
+                isclose(0, source_pixels[i][0], abs_tol=1))
+            assert isclose(source_pixels[i][1], output_pixels[i][1], abs_tol=5) or (
+                isclose(0, source_pixels[i][1], abs_tol=1))
+            assert isclose(source_pixels[i][2], output_pixels[i][2], abs_tol=5) or (
+                isclose(0, source_pixels[i][2], abs_tol=1))
+
+    def test_bmp(self, directory):
+        source_file_path = os.path.join(BASE, 'files', 'test.bmp')
+        output_file_path = os.path.join(directory, 'test.{}'.format(settings.EXPORT_TYPE))
+        format = '{}.{}'.format(settings.EXPORT_MAXIMUM_SIZE, settings.EXPORT_TYPE)
+        exporter = ImageExporter(source_file_path=source_file_path, ext='.bmp',
+                            output_file_path=output_file_path, format=format)
+
+        assert not os.path.exists(output_file_path)
+        assert os.path.exists(directory)
+
+        exporter.export()
+
+        assert os.path.exists(output_file_path)
+
+        output_image = Image.open(output_file_path)
+        source_image = Image.open(source_file_path)
+        output_pixels = list(output_image.getdata())
+
+        assert source_image.size == output_image.size
+        assert output_image.mode == 'RGB'
+        assert output_image.palette == source_image.palette
+        assert output_image.format.lower() == settings.EXPORT_TYPE
+        # looped pixel tests not included because values varried so much that
+        # they were basically useless
+        assert output_pixels[0] == (255, 254, 232)
+
+    def test_ratio(self, directory):
+        source_file_path = os.path.join(BASE, 'files', 'test_ratio.png')
+        output_file_path = os.path.join(directory, 'test.{}'.format(settings.EXPORT_TYPE))
+        format = '{}.{}'.format(settings.EXPORT_MAXIMUM_SIZE, settings.EXPORT_TYPE)
+        exporter = ImageExporter(source_file_path=source_file_path, ext='.png',
+                            output_file_path=output_file_path, format=format)
+
+        assert not os.path.exists(output_file_path)
+        assert os.path.exists(directory)
+
+        exporter.export()
+
+        assert os.path.exists(output_file_path)
+
+        output_image = Image.open(output_file_path)
+        source_image = Image.open(source_file_path)
+
+        assert output_image.mode == 'RGB'
+        assert output_image.size == (1200, 1029)
+        assert output_image.palette == source_image.palette
+        assert output_image.format.lower() == settings.EXPORT_TYPE
+
+    def test_exceptions(self, directory):
```

Comment:
Re look over this and figure out which error it is testing and make a note

also probably add test for not an image file. so if you pass in a corrupted image, it raises the correct exception.

Reply:
also try is you rename a .png to .jpg and see what happens. (rename image to different image type and see if pillow opens it)

Reply:
Completed error labeling. completed add corrupted image file. renaming pngs to and from jpg did nothing, so skipping that.